controllers/awsmachine_controller.go

```diff
@@ -197,10 +199,16 @@ func (r *AWSMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	infrav1.SetDefaults_AWSMachineSpec(&awsMachine.Spec)
 
+	cp, err := r.getControlPlane(ctx, log, cluster)
```

Comment:
I'm sorry, I'm probably missing something, but how is this related change? 🤔

Reply:
The Control Plane object is required since the Cluster API contract to determine if it's externally managed resides on its status (`/status/externalManagedControlPlane` boolean value).

Without it, we cannot understand if the scope is actually `IsControlPlaneExternallyManaged` or not.

Reply:
Understood. Thanks for the explanation!